pkg/rbd/controllerserver.go

```diff
@@ -177,6 +185,14 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 		return nil, err
 	}
 
+	if rbdVol.Encrypted {
+		err = ensureEncryptionMetadataSet(ctx, cr, rbdVol)
+		if err != nil {
+			klog.Errorf(util.Log(ctx, err.Error()))
```

Comment:
we need to delete the rbd image if medataset fails, as  we  should not leave stale rbd image

Reply:
done